### DISCUSSION TITLE
How to pinning 2 columns?

### DISCUSSION THREAD

**Discussion by ChanoPark** [KevinVandy/material-react-table#213] [Q&A] [answered]
hello!
I am using your library well.
Thank you.

I want to pin multiple columns. (I want to anchor multiple columns to the left like in your document.)
<img width="421" alt="image" src="https://user-images.githubusercontent.com/77626299/203486525-2c290a51-5703-4158-9bc8-cd97775b2a67.png">

I wanted to fix the expand button and the first column, so I tried the following.
![image](https://user-images.githubusercontent.com/77626299/203486802-406904a0-a1ac-4432-9011-f4022d237915.png)

But the expand button is covered by the first column. Perhaps, it is because of the relatively low z-index.

How can I move the expand button and the first column together?

I have additional questions.
- How can I indent when a row is expanded? The size is fixed so it doesn't seem to be indented.

Thanks again.

**Comment by KevinVandy** [comment, marked as answer]
You might just need to make the column wider with the `size` option on a column. Otherwise, share a screenshot for what problem you are running into.

**Reply by ChanoPark** [reply]
### 1. Problem that expand button and pinned column do not move together when scrolling.

When first rendered, it looks like this:
![image](https://user-images.githubusercontent.com/77626299/203671775-d162d7f2-1302-4700-8086-c2d545f72292.png)

But when I try to scroll right, the expand button is hidden by the group column.
![image](https://user-images.githubusercontent.com/77626299/203671941-d4371118-8b7e-4cd6-89a9-66f20304db6b.png)

raising the z-index of the cell containing the expand button in Chrome Developer Tools gives:
![스크린샷 2022-11-24 오전 10 11 27](https://user-images.githubusercontent.com/77626299/203672413-4ca83fea-2be4-4410-9643-92577d668deb.png)

So, I think the expand button and the fixed column(group column) move together.  
However, it seems that the fixed position is not correct. How can this be fixed?

### 2. Problem that sub row is not indented.
Expanding the row looks like this:
![image](https://user-images.githubusercontent.com/77626299/203672755-7b88e1c9-a9ba-4528-9ae8-4c1581d0e826.png)

I receive the data dynamically, so I can't randomly add spaces to the data.
Is there a way to indent only the content while the table size is fixed?

I'm not familiar with react, but I was able to solve the problem thanks to your good library.
Thanks you :)

**Reply by KevinVandy** [reply]
Using the `displayColumnDefOptions` prop, you can change the min-width (minSize) of expand column, or even it's z-index if you want.

https://www.material-react-table.com/docs/guides/display-columns#customize-built-in-mrt-display-columns

**Reply by ChanoPark** [reply]
I tried the following:
![image](https://user-images.githubusercontent.com/77626299/204171339-1a6bed07-f857-42e1-839d-f24f098f39b8.png)

It works fine if I change only the z-index without changing the size. But when I set the size it doesn't work. (It does not become a z-index setting.)
I want the cell to be as big as the arrow (max-content).

Without resizing it would look like this:
<img width="75" alt="image" src="https://user-images.githubusercontent.com/77626299/204171674-93eade9b-a743-4feb-8b53-898a1de049aa.png">

I tried various things, but could not get an answer.
I'm sorry to bother you. T.T

**Reply by KevinVandy** [reply]
Use integers for size. The unit is pixels.

**Reply by KevinVandy** [reply]
Also zIndex would have to be in an sx prop since it is CSS.

**Reply by ChanoPark** [reply]
I think the z-index doesn't seem to apply if the size is smaller than the expand button.
I solved it by setting size to 25. Thank you for your kind reply.

**Comment by ingegneri-bruno-clorox** [comment]
Yes, I was able to make it pin 4 columns to the left even using `layout: semantic` but, without the auto-sizing feature for the column. It seems that these above solutions for multiple left column pinning does not cover the scenario where I want to use both `layout: semantic` for table configuration and `size: auto` for the columns.

_Just to clarify, I call it auto-sizing when I explicitly declare `size: auto` for the column. Without this parameter, it automatically sets a default column width, but it does not make the best use of the available space._

_And I did not want to open a brand-new discussion for it because it falls under similar issues, and keeping it together would facilitate other users finding the best approach for their cases._

**Is there a solution to combine both multiple column pinning and column auto-sizing?**
-------------------------
